src/test/fuzz/tx_pool.cpp

```diff
@@ -177,45 +178,54 @@ FUZZ_TARGET_INIT(tx_pool_standard, initialize_tx_pool)
-            tx_mut.nLockTime = fuzzed_data_provider.ConsumeBool() ? 0 : fuzzed_data_provider.ConsumeIntegral<uint32_t>();
-            const auto num_in = fuzzed_data_provider.ConsumeIntegralInRange<int>(1, outpoints_rbf.size());
-            const auto num_out = fuzzed_data_provider.ConsumeIntegralInRange<int>(1, outpoints_rbf.size() * 2);
-
-            CAmount amount_in{0};
-            for (int i = 0; i < num_in; ++i) {
-                // Pop random outpoint
-                auto pop = outpoints_rbf.begin();
-                std::advance(pop, fuzzed_data_provider.ConsumeIntegralInRange<size_t>(0, outpoints_rbf.size() - 1));
-                const auto outpoint = *pop;
-                outpoints_rbf.erase(pop);
-                amount_in += GetAmount(outpoint);
-
-                // Create input
-                const auto sequence = ConsumeSequence(fuzzed_data_provider);
-                const auto script_sig = CScript{};
-                const auto script_wit_stack = std::vector<std::vector<uint8_t>>{WITNESS_STACK_ELEM_OP_TRUE};
-                CTxIn in;
-                in.prevout = outpoint;
-                in.nSequence = sequence;
-                in.scriptSig = script_sig;
-                in.scriptWitness.stack = script_wit_stack;
-
-                tx_mut.vin.push_back(in);
-            }
-            const auto amount_fee = fuzzed_data_provider.ConsumeIntegralInRange<CAmount>(-1000, amount_in);
-            const auto amount_out = (amount_in - amount_fee) / num_out;
-            for (int i = 0; i < num_out; ++i) {
-                tx_mut.vout.emplace_back(amount_out, P2WSH_OP_TRUE);
-            }
-            const auto tx = MakeTransactionRef(tx_mut);
-            // Restore previously removed outpoints
-            for (const auto& in : tx->vin) {
-                Assert(outpoints_rbf.insert(in.prevout).second);
+        const Package package = [&] {
+            auto n_tx = fuzzed_data_provider.ConsumeIntegralInRange<uint32_t>(1, MAX_PACKAGE_COUNT + 5);
+            Package txs;
+            while (n_tx--) {
+                CMutableTransaction tx_mut;
+                tx_mut.nVersion = CTransaction::CURRENT_VERSION;
+                tx_mut.nLockTime = fuzzed_data_provider.ConsumeBool() ? 0 : fuzzed_data_provider.ConsumeIntegral<uint32_t>();
+                const auto num_in = fuzzed_data_provider.ConsumeIntegralInRange<int>(1, outpoints_rbf.size());
+                const auto num_out = fuzzed_data_provider.ConsumeIntegralInRange<int>(1, outpoints_rbf.size() * 2);
+
+                CAmount amount_in{0};
+                for (int i = 0; i < num_in; ++i) {
+                    // Pop random outpoint
+                    auto pop = outpoints_rbf.begin();
+                    std::advance(pop, fuzzed_data_provider.ConsumeIntegralInRange<size_t>(0, outpoints_rbf.size() - 1));
+                    const auto outpoint = *pop;
+                    outpoints_rbf.erase(pop);
+                    amount_in += GetAmount(outpoint);
```

Comment:
Now the fuzz target has some shallow bugs. For example, `GetAmount` fail to get coin value. I guess that's because I simply add newly constructed tx out to outpoints, so `CCoinsViewMemPool::GetCoin` cannot find the outpoint in mempool.

```log
test/fuzz/tx_pool.cpp:164 operator(): Assertion `amount_view.GetCoin(outpoint, c)' failed.
==3026798== ERROR: libFuzzer: deadly signal
    #0 0x5609ccdba680 in __sanitizer_print_stack_trace (/home/ubuntu/bitcoin/src/test/fuzz/fuzz+0xfd7680)
    #1 0x5609ccd66988 in fuzzer::PrintStackTrace() (/home/ubuntu/bitcoin/src/test/fuzz/fuzz+0xf83988)
    #2 0x5609ccd4bad3 in fuzzer::Fuzzer::CrashCallback() (/home/ubuntu/bitcoin/src/test/fuzz/fuzz+0xf68ad3)
    #3 0x7f173207341f  (/lib/x86_64-linux-gnu/libpthread.so.0+0x1441f)
    #4 0x7f1731cd900a in __libc_signal_restore_set /build/glibc-SzIz7B/glibc-2.31/signal/../sysdeps/unix/sysv/linux/internal-signals.h:86:3
    #5 0x7f1731cd900a in raise /build/glibc-SzIz7B/glibc-2.31/signal/../sysdeps/unix/sysv/linux/raise.c:48:3
    #6 0x7f1731cb8858 in abort /build/glibc-SzIz7B/glibc-2.31/stdlib/abort.c:79:7
    #7 0x5609cf0fa49d in assertion_fail(char const*, int, char const*, char const*) src/util/check.cpp:13:5
    #8 0x5609cd319c02 in bool&& inline_assertion_check<true, bool>(bool&&, char const*, int, char const*, char const*) src/./util/check.h:67:13
    #9 0x5609cd5cbdda in (anonymous namespace)::tx_pool_standard_fuzz_target(Span<unsigned char const>)::$_4::operator()(COutPoint const&) const src/test/fuzz/tx_pool.cpp:164:9
    #10 0x5609cd5cc62a in (anonymous namespace)::tx_pool_standard_fuzz_target(Span<unsigned char const>)::$_5::operator()() const src/test/fuzz/tx_pool.cpp:198:34
```

Reply:
I find changing `amount_view` to be non-const and calling `amount_view.PackageAddTransaction(tx);` after each tx is constructed can also fix the assert.